tobac/segmentation.py

```diff
@@ -32,6 +32,181 @@
 
 import logging
 
+import skimage
+
+from . import utils as tb_utils
+from .utils import internal as internal_utils
+
+
+def add_markers(features, marker_arr, seed_3D_flag, seed_3D_size=5, level=None):
+    """Adds markers for watershedding using the `features` dataframe
+    to the marker_arr.
+
+    Parameters
+    ----------
+    features: pandas.DataFrame
+        Features for one point in time to add as markers.
+    marker_arr: 2D or 3D array-like
+        Array to add the markers to. Assumes a (z, h1, h2) or (h1, h2) configuration.
+    seed_3D_flag: str('column', 'box')
+        Seed 3D field at feature positions with either the full column
+         or a box of user-set size
+    seed_3D_size: int or tuple (dimensions equal to dimensions of `field`)
+        This sets the size of the seed box when `seed_3D_flag` is 'box'. If it's an
+        integer, the seed box is identical in all dimensions. If it's a tuple, it specifies the
+        seed area for each dimension separately.
+        Note: we recommend the use of odd numbers for this. If you give
+        an even number, your seed box will be biased and not centered
+        around the feature.
+        Note: if two seed boxes overlap, the feature that is seeded will be the
+        closer feature.
+    level: slice or None
+        If `seed_3D_flag` is 'column', the levels at which to seed the
+        cells for the watershedding algorithm. If None, seeds all levels.
+
+    Returns
+    -------
+    2D or 3D array like (same type as `marker_arr`)
+        The marker array
+    """
+    import numpy as np
+
+    # What marker number is the background? Assumed 0.
+    bg_marker = 0
+
+    if level is None:
+        level = slice(None)
+
+    if len(marker_arr.shape) == 3:
+        is_3D = True
+        z_len = marker_arr.shape[0]
+
+    else:
+        is_3D = False
+        z_len = 0
+        # transpose to 3D array to make things easier.
+        marker_arr = marker_arr[np.newaxis, :, :]
+
+    if seed_3D_flag == "column":
+        for index, row in features.iterrows():
+            marker_arr[level, int(row["hdim_1"]), int(row["hdim_2"])] = row["feature"]
+
+    elif seed_3D_flag == "box":
```

Comment:
Should raise an exception if seed_3D_flag is neither "column" nor "box"

Reply:
Good point, I've added this check up front.